### PR TITLE
feat: adding ability to use a separate bounding box for resizing and dragging (Fixes #895)

### LINKED ISSUE #895: Is It Possible To Set Bound Dynamically?
### I Have A Question
Is it possible to set an initial boundary like "parent" to limit resizing, but, when dragging, change that bound to a bigger container?

I'm using react-rnd 10.3.7
My browser is Chrome Version 107.0.5304.122 (Official Build) (64-bit)
I could not find any reported issues that describe what I'm asking.
   
### Reproduced project

https://codesandbox.io/s/y3997qply9

### Description
I have several divs each with a class of "page" (dark blue solid border) and they are all contained within a single scrolling div (green dashed border). Within a given "page" div I have several <Rnd> components (one is shown with a pale blue border and yellow background), each with a bound prop set to 'parent' which limits their sizes to be entirely within the "page" boundary. This works perfectly. 

What I'd like to do next is be able to drag any of these <Rnd> to other pages. This would require me to somehow break out of the current parent boundary so that it can be allowed to cross over to another page. The restriction that I have is that it cannot be dropped between pages (i.e., into the red zone between pages). Obviously the bound of "parent" is currently preventing this desired functionality, but, at the same time it is working to limit the sizing of the <Rnd> to be within the page.

Can this be done with a "dynamic" bound somehow? Any feedback and suggestions are greatly appreciated.

image

### PR DESCRIPTION
**Pull request by branks**
This patch has been running in production for us for over a year and works well. It allows the resize and drag boundarys to be different. I thought I would convert the patch into a PR to share with others

I was unable to run the tests locally so couldn't add anymore - there's an error when running 

```
.../react-rnd/node_modules/electron/dist/electron: error while loading shared libraries: libgconf-2.so.4: cannot open shared object file: No such file or directory
```

but this is the work required

### CODE REVIEW

**Comment by bokuweb** [comment]
@branks Thanks for your PR.

File: src/index.tsx [hunk 1]
```diff
@@ -130,6 +130,7 @@ export interface Props {
   size?: Size;
   resizeGrid?: Grid;
   bounds?: string | Element;
+  resizeBounds?: string | Element;
```
Comment (bokuweb) [inline src/index.tsx:133]: I want to minimize the number of props as much as possible.
Can the issue be addressed by expanding the `bounds` as follows?

``` TypeScript
bounds?: (string | Element) | { type: 'resize': bounds: string | Element } |   { type: 'drag': bounds: string | Element } 
```

**Comment by bokuweb** [comment]
Could you please add story and fix README too?

File: src/index.tsx (hunk 1 above)
Comment (branks) [reply src/index.tsx:133]: So there's times where I want to pass both bounds to the object 

This means bounds then has to either become an array with both bounds types in it, or an object with keys of resize and drag, which do you prefer?

Comment (bokuweb) [reply src/index.tsx:133]: I see. Please wait a moment.

Comment (branks) [reply src/index.tsx:133]: OK great, I'll try and get round to this in the next few days, thanks

**Comment by branks** [comment]
Closing as I dont have the time to update this - someone else is welcome to pick it up - sorry!

**Pull request closed without merging**